examples/message-ubench/main.pony

```diff
@@ -283,7 +321,19 @@ class Tick is TimerNotify
     _env = env
     _sync_leader = sync_leader
 
-    fun ref apply(timer: Timer, count: U64): Bool =>
-      _sync_leader.tick_fired(count)
-      true
+  fun ref apply(timer: Timer, count: U64): Bool =>
+    _sync_leader.tick_fired(count)
+    true
+
+class RunTimerNotify is TimerNotify
+  let _env: Env
+  let _sync_leader: SyncLeader
+
+  new iso create(env: Env, sync_leader: SyncLeader) =>
+    _env = env
+    _sync_leader = sync_leader
+
+  fun ref apply(timer: Timer, count: U64): Bool =>
+    _sync_leader.run_done(count)
+    false
```

Comment:
by returning false here, the timer won't be rearmed so there's no need to cancel elsewhere.